File: answers/lintcode/Unique Paths.cpp
class Solution {
public:
    /**
     * @param n, m: positive integer (1 <= n ,m <= 100)
     * @return an integer
     */
    int uniquePaths(int m, int n) {
        // wirte your code here
        vector<vector<int> > path_list(m, vector<int> (n, 0));
        for (size_t i = 0; i < path_list.size(); ++ i) {
            for (size_t j = 0; j < path_list[i].size(); ++ j) {
                if (0 == i && 0 == j) {
                    path_list[i][j] = 1;
                }
                if (i > 0) {
                    path_list[i][j] += path_list[i - 1][j];
                }
                if (j > 0) {
                    path_list[i][j] += path_list[i][j - 1];
                }
            }
        }
        return path_list.back().back();
    }
};

File: answers/leetcode/Unique Paths.py
class Solution(object):
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """

        dp = [[0 for i in range(n)] for j in range(m)]
        dp[0][0] = 1
        for i in range(m):
            for j in range(n):
                dp[i][j] += (dp[i - 1][j] if i > 0 else 0) + (dp[i][j - 1] if j > 0 else 0)
        return dp[-1][-1]
